packages/core/src/mastra/cross-process-workflow.test.ts

```
@@ -0,0 +1,206 @@
+ *
+ * When two Mastra instances share a push-only pubsub (mimicking Unix socket
+ * IPC between mc processes), events for internal workflows (execution-workflow,
+ * agentic-loop) registered on one instance must NOT be processed by the other.
+ * Without the guard the WEP would call errorWorkflow() → publish workflow.fail
+ * → processWorkflowFail → workflows-finish, erroneously terminating the
+ * correct instance's run.
+ */
+import { describe, expect, it, vi } from 'vitest';
+import { z } from 'zod/v4';
+
+import { createStep, createWorkflow } from '../workflows/evented';
```

ESLint

[error] `../workflows/evented` import should occur after import of `../storage/mock`

(import/order)